dtls.c

```diff
@@ -2766,18 +2766,14 @@ check_server_hello(dtls_context_t *ctx,
 {
   dtls_handshake_parameters_t *handshake = peer->handshake_params;
 
-  /* This function is called when we expect a ServerHello (i.e. we
-   * have sent a ClientHello).  We might instead receive a HelloVerify
-   * request containing a cookie. If so, we must repeat the
-   * ClientHello with the given Cookie.
-   */
-  if (data_length < DTLS_HS_LENGTH + DTLS_HS_LENGTH)
+  /* Check we have enough data for the ServerHello */
+  if (data_length < DTLS_HS_LENGTH + 2 + DTLS_RANDOM_LENGTH + 1 + 2 + 1) {
```

Comment:
This test solves only part of the issue as the session id might not be empty. To handle that case, the data access after the `SKIP_VAR_FIELD()` for the session id must be protected as well.

Reply:
Good catch.  Length rechecked after `SKIP_VAR_FIELD()` and code pushed.